builtin/providers/google/resource_compute_router_test.go

```diff
@@ -0,0 +1,176 @@
+package google
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform/helper/acctest"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+
+	"google.golang.org/api/compute/v1"
+)
+
+func TestAccComputeRouter_basic(t *testing.T) {
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckComputeRouterDestroy,
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config: testAccComputeRouter_basic,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckComputeRouterExists(
+						"google_compute_router.foobar"),
+					resource.TestCheckResourceAttr(
+						"google_compute_router.foobar", "region", "europe-west1"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccComputeRouter_noRegion(t *testing.T) {
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckComputeRouterDestroy,
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config: testAccComputeRouter_noRegion,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckComputeRouterExists(
+						"google_compute_router.foobar"),
+					resource.TestCheckResourceAttr(
+						"google_compute_router.foobar", "region", "us-central1"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccComputeRouter_networkLink(t *testing.T) {
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckComputeRouterDestroy,
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config: testAccComputeRouter_networkLink,
+				Check: testAccCheckComputeRouterExists(
+					"google_compute_router.foobar"),
+			},
+		},
+	})
+}
+
+func testAccCheckComputeRouterDestroy(s *terraform.State) error {
+	config := testAccProvider.Meta().(*Config)
+	project := config.Project
+
+	routersService := compute.NewRoutersService(config.clientCompute)
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "google_compute_router" {
+			continue
+		}
+
+		region := rs.Primary.Attributes["region"]
```

Comment:
Would like to see this support the same `getRegion` logic the resource does, to avoid future bugs.